### PR TITLE
Adding SqlJson and Data write with SqlJson (#2752)

### PR DESCRIPTION
**Pull request by deepaksa1**
This PR introduces the `SqlJson` class along with its test.

##Constructors
`SqlJson()` Creates a new SqlJson instance.
`SqlJson(string)` Takes a string as input and initializes a new instance of the SqlJson class
`SqlJson(JsonDocument)` Takes a JsonDocument as input and initializes a new instance of the SqlJson

##Properties
`IsNull` Indicates whether this instance represents a null SqlJson value.
`Null` Represents a null instance of the SqlJson type
`Value` Gets the string representation of the Json content of this SqlJson instance

##Methods
`SqlDataReader.GetSqlJson(int)` Retrieves the column at ordinal as a SqlJson

### CODE REVIEW

**Review by apoorvdeshmukh — COMMENTED** [review]
@deepaksa1 Can you check the test failures?

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDataReader.cs [hunk 1]
```diff
@@ -2894,6 +2895,19 @@ virtual public SqlXml GetSqlXml(int i)
             return sx;
         }
 
+        /// <summary>
+        /// Retrieves the column at ordinal as a SqlJson.
+        /// </summary>
+        /// <param name="i"></param>
+        /// <returns></returns>
+        virtual public SqlJson GetSqlJson(int i)
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDataReader.cs:2903]: Please move documentation to XML file and also include this API + doc in ref projects.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs [hunk 2]
```diff
@@ -2541,6 +2542,18 @@ virtual public SqlXml GetSqlXml(int i)
             return sx;
         }
 
+        /// <summary>
+        /// Retrieves the column at ordinal as a SqlJson.
+        /// </summary>
+        /// <param name="i"></param>
+        /// <returns></returns>
+        virtual public SqlJson GetSqlJson(int i)
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs:2550]: Same here as well.

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs [hunk 3]
```diff
@@ -937,6 +939,22 @@ internal SqlString SqlString
...
+                {
+                    if (IsNull)
+                    { 
+                        return SqlTypes.SqlJson.Null;
+                    }
+                    return new SqlJson((string)_object);
+                }
+                return (SqlJson)SqlValue;
+            }
+        }
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs:956]:
```suggestion
        internal SqlJson SqlJson => (StorageType.Json == _type) ? (IsNull ? SqlTypes.SqlJson.Null : new SqlJson((string)_object)) : (SqlJson)SqlValue;
```

**Comment by cheenamalhotra** [comment]
Can you update PR description to include new APIs added to the driver?

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs (hunk 3 above)
Comment (deepaksa1) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs]: Done

**Review by apoorvdeshmukh — APPROVED** [review]
Looks Good. Do update the PR description with API details as suggested by @cheenamalhotra

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**